onnxscript/rewriter/function_rule.py

```diff
@@ -128,24 +127,14 @@ def _find_node_contains_key_in_name(
         return None
 
     def _find_node_by_type(
```

Comment:
@justinchuby Do we have a better approach for node searching?

Reply:
Nothing better in the IR. Would it be helpful to build an auxiliary data structure here? Eg a dictionary?

Reply:
I think we can extend it when we have other needs.